cmd/marker/main.go

```diff
@@ -137,3 +115,68 @@ func keepAlive(healthCheckFile string, healthCheckInterval int) {
 
 	}, time.Duration(healthCheckInterval)*time.Second)
 }
+
+func parseOvsSocket(ovsSocket *string) (string, error) {
```

Comment:
> I won't fight for it, but since you are so kind to contribute this PR, I will try:
> 
> Would you please split this function into multiple functions? I think that the part where we parse and the one where you attempt to connect could easily be two different ones. And perhaps the connection test could call two different functions functions, one for unix, one for tcp.

Totally agree, done as well.